### PR TITLE
Change validated authorities look up

### PR DESCRIPTION
**Pull request by unpluggedk**
#125 

Change validated authorities to resolved authorities, and add validated look up upon retrieval.

### CODE REVIEW

File: MSAL/src/instance/MSALAuthority.h [hunk 1]
```diff
@@ -60,7 +60,7 @@ typedef void(^MSALAuthorityCompletion)(MSALAuthority *authority, NSError *error)
 
 @property MSALAuthorityType authorityType;
 @property NSURL *canonicalAuthority;
-@property BOOL validateAuthority;
+@property BOOL validatedAuthority;
```
Comment (partnerinflight) [inline MSAL/src/instance/MSALAuthority.h]: nit: if this is meant to represent an authority that's been validated, then maybe "isValidatedAuthority" would be clearer?

File: MSAL/src/instance/MSALAuthority.m [hunk 2]
```diff
@@ -163,8 +163,12 @@ + (void)resolveEndpointsForAuthority:(NSURL *)unvalidatedAuthority
     
     if (authorityInCache)
     {
-        completionBlock(authorityInCache, nil);
-        return;
+        if (!validate ||
```
Comment (partnerinflight) [inline MSAL/src/instance/MSALAuthority.m]: q: is validate expected to be set externally? Or where is it coming from?

File: MSAL/src/instance/MSALAuthority.h (hunk 1 above)
Comment (RPangrle) [inline MSAL/src/instance/MSALAuthority.h]: In objC style "isValidatedAuthority" would be the name of the accessor for a BOOL of validatedAuthority

File: MSAL/src/instance/MSALAuthority.m (hunk 2 above)
Comment (RPangrle) [inline MSAL/src/instance/MSALAuthority.m]: It's coming from the validateAuthority property on the PublicClientApplication

File: MSAL/src/instance/MSALAuthority.m [hunk 3]
```diff
@@ -41,8 +41,8 @@ @implementation MSALAuthority
 #define TENANT_ID_STRING_IN_PAYLOAD @"{tenantid}"
 
 static NSSet<NSString *> *s_trustedHostList;
-static NSMutableDictionary *s_validatedAuthorities;
-static NSMutableDictionary *s_validatedUsersForAuthority;
+static NSMutableDictionary *s_resolvedAuthorities;
+static NSMutableDictionary *s_reslovedUsersForAuthority;
```
Comment (oldalton) [inline MSAL/src/instance/MSALAuthority.m]: nit: There's a typo "resloved"

File: MSAL/src/instance/MSALAuthority.m (hunk 2 above)
Comment (oldalton) [inline MSAL/src/instance/MSALAuthority.m]: I realized it wasn't actually being taken from there (or from anywhere else). This should fix it: #149

**Pull request merged**